src/test/java/io/camunda/testing/assertions/ProcessInstanceAssertionsTest.java

```diff
@@ -0,0 +1,710 @@
+    }
+
+    @Test
+    public void testProcessInstanceIsWaitingForMessageFailure() throws InterruptedException {
+      // given
+      deployProcess(MESSAGE_EVENT_BPMN);
+      final String correlationKey = "key";
+      final Map<String, Object> variables =
+          Collections.singletonMap("correlationKey", correlationKey);
+      final ProcessInstanceEvent instanceEvent =
+          startProcessInstance(MESSAGE_EVENT_PROCESS_ID, variables);
+
+      // when
+      sendMessage(MESSAGE_NAME, correlationKey);
+
+      // then
+      assertThatThrownBy(() -> assertThat(instanceEvent).isWaitingForMessage(MESSAGE_NAME))
+          .isInstanceOf(AssertionError.class)
+          .hasMessage(
+              "Process with key %s is not waiting for message(s) with name(s) %s",
+              instanceEvent.getProcessInstanceKey(), MESSAGE_NAME);
+    }
+
+    @Test
+    public void testProcessInstanceIsNotWaitingForMessageFailure() throws InterruptedException {
+      // given
+      deployProcess(MESSAGE_EVENT_BPMN);
+      final String correlationKey = "key";
+      final Map<String, Object> variables =
+          Collections.singletonMap("correlationKey", correlationKey);
+
+      // when
+      final ProcessInstanceEvent instanceEvent =
+          startProcessInstance(MESSAGE_EVENT_PROCESS_ID, variables);
+
+      // then
+      assertThatThrownBy(() -> assertThat(instanceEvent).isNotWaitingForMessage(MESSAGE_NAME))
+          .isInstanceOf(AssertionError.class)
+          .hasMessage(
+              "Process with key %s is waiting for message(s) with name(s) %s",
+              instanceEvent.getProcessInstanceKey(), MESSAGE_NAME);
+    }
+  }
+
+  private void deployProcess(final String process) {
+    client.newDeployCommand().addResourceFromClasspath(process).send().join();
+  }
+
+  private ProcessInstanceEvent startProcessInstance(final String processId)
+      throws InterruptedException {
+    return startProcessInstance(processId, new HashMap<>());
+  }
+
+  private ProcessInstanceEvent startProcessInstance(
+      final String processId, final Map<String, Object> variables) throws InterruptedException {
+    final ProcessInstanceEvent instanceEvent =
+        client
+            .newCreateInstanceCommand()
+            .bpmnProcessId(processId)
+            .latestVersion()
+            .variables(variables)
+            .send()
+            .join();
+    Thread.sleep(100);
+    return instanceEvent;
+  }
+
+  // TODO we need a proper way to complete jobs instead of this hack
+  private void completeTask(final String elementId) throws InterruptedException {
+    Thread.sleep(100);
+    Record<JobRecordValue> lastRecord = null;
+    for (Record<JobRecordValue> record : engine.jobRecords().withElementId(elementId)) {
+      if (record.getIntent().equals(JobIntent.CREATED)) {
+        lastRecord = record;
+      }
+    }
+    if (lastRecord != null) {
+      client.newCompleteCommand(lastRecord.getKey()).send().join();
```

Comment:
:clown_face: I am surprised that this works at all, tbh. Not sure if this is a bug or a feature :thinking:

Reply:
❓ Why would this not work? I send a complete command with a jobkey. I'm not sure what the bug would be.

Reply:
Maybe I am missing something. Who activated the job? To me it looks like this is completing a job before it has been activated.

Reply:
Ooh right. @Zelldon at some point mentioned to me that it is not required to activate a job before completing it. I'm not sure if it is intended or not.

Reply:
Good to know :-)